core/trino-main/src/main/java/io/trino/operator/HashGenerator.java

```diff
@@ -24,6 +24,13 @@ default int getPartition(int partitionCount, int position, Page page)
         long rawHash = hashPosition(position, page);
         // This function reduces the 64 bit rawHash to [0, partitionCount) uniformly. It first reduces the rawHash to 32 bit
         // integer x then normalize it to x / 2^32 * partitionCount to reduce the range of x from [0, 2^32) to [0, partitionCount)
-        return (int) ((Integer.toUnsignedLong(Long.hashCode(rawHash)) * partitionCount) >>> 32);
+        return partitionUniformly(Long.hashCode(rawHash), partitionCount);
+    }
+
+    // This function reduces the 32 bit rawHash to [0, partitionCount) uniformly.
+    // It normalizes rawHash to x / 2^32 * partitionCount to reduce the range of x from [0, 2^32) to [0, partitionCount)
+    static int partitionUniformly(int hash, int partitionCount)
```

Comment:
This seems like the wrong place for this utility method. Not sure where it should go, but making `HashGenerator` interface a utility repository seems off.

BTW, there's nothing magic about this function -- it's a simple normalization function. It computes `x/MAX_X * N`. So, while `partitionUniformly` describes how this formula is being used, it doesn't describe what the function is doing. Maybe we should move it to MoreMath and rename it accordingly.

Reply:
I moved this method to `MoreMath` but for now, I left the `partitionUniformly` name.
I think `partitionUniformly` better conveys what's the purpose/contract and if I named it let's say `normalizeTo` or `scaleDownTo` it would better describe the implementation (i.e. how it's doing it).
My preference is the former.